### PR TITLE
Add table aws_emr_cluster_metric_isidle closes #554

### LINKED ISSUE #554: Add table aws_emr_cluster_metric_isidle
**References**

https://aws.amazon.com/blogs/big-data/optimize-amazon-emr-costs-with-idle-checks-and[…]tion-using-advanced-amazon-cloudwatch-metrics-and-aws-lambda/

Use case:
Detect clusters that are idle . The IsIdle metric tracks whether a cluster is live, but not currently running tasks. You can set an alarm to fire when the cluster has been idle for a given period of time, such as thirty minutes.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select * from aws_emr_cluster_metric_idle
+-----------------+-------------+----------------------+---------+---------+---------+--------------+-----+------+---------------------+-----------+-----------+--------------+
| cluster_name    | metric_name | namespace            | average | maximum | minimum | sample_count | sum | unit | timestamp           | partition | region    | account_id   |
+-----------------+-------------+----------------------+---------+---------+---------+--------------+-----+------+---------------------+-----------+-----------+--------------+
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:16:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:36:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:21:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:46:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:16:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:36:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:31:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:26:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:56:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:11:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:51:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:16:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:11:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:21:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:31:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:51:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:01:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:01:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:16:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:46:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:51:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:41:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:21:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:06:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:41:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:31:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:11:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:01:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:46:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:26:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:31:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:21:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:41:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:56:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:41:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:56:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:06:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:46:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:11:00 | aws       | us-east-2 | 986325076436 |
| j-1JKN67QI4PEJS | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-22 13:11:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:36:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:01:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:56:00 | aws       | us-east-2 | 986325076436 |
| j-3NYO8BYCDSZI7 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-22 12:56:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:51:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:11:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:21:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:46:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:06:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:36:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 10:26:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:36:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:31:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:26:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 08:16:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:51:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 09:06:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 07:26:00 | aws       | us-east-2 | 986325076436 |
| j-27072W1DU3OT4 | IsIdle      | AWS/ElasticMapReduce | 1       | 1       | 1       | 1            | 1   | None | 2021-07-26 06:41:00 | aws       | us-east-2 | 986325076436 |
+-----------------+-------------+----------------------+---------+---------+---------+--------------+-----+------+---------------------+-----------+-----------+--------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_emr_cluster_metric_idle.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package aws
```
Comment (bigdatasourav) [inline aws/table_aws_emr_cluster_metric_idle.go:1]: please update the file name.

File: aws/table_aws_emr_cluster_metric_idle.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsEmrClusterMetricIsidle(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_emr_cluster_metric_isidle",
+		Description: "AWS EMR Cluster Cloudwatch Metrics - IsIdle",
```
Comment (bigdatasourav) [inline aws/table_aws_emr_cluster_metric_idle.go:17]:
```suggestion
		Description: "AWS EMR Cluster Cloudwatch Metrics - Isidle",
```

File: aws/plugin.go [hunk 3]
```diff
@@ -137,6 +137,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_elasticache_subnet_group":                                tableAwsElastiCacheSubnetGroup(ctx),
 			"aws_elasticsearch_domain":                                    tableAwsElasticsearchDomain(ctx),
 			"aws_emr_cluster":                                             tableAwsEmrCluster(ctx),
+			"aws_emr_cluster_metric_isidle":                               tableAwsEmrClusterMetricIsidle(ctx),
```
Comment (bigdatasourav) [inline aws/plugin.go:140]:
```suggestion
			"aws_emr_cluster_metric_isidle":                               tableAwsEmrClusterMetricIsIdle(ctx),
```

File: aws/table_aws_emr_cluster_metric_idle.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_emr_cluster_metric_idle.go]: Updated the file name

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**